fabtests/ubertest/uber.c

```diff
@@ -652,82 +480,125 @@ static int ft_fw_server(void)
 
 		results[ft_fw_result_index(ret)]++;
 
-	} while (!ret || ret == FI_EIO || ret == FI_ENODATA);
+	} while (!ret || ret == -FI_EIO || ret == -FI_ENODATA || ret == -FT_SKIP);
 
 	return ret;
 }
-
-static int ft_client_child(void)
+static int ft_client_setup(struct fi_info *hints, struct fi_info *info)
 {
-	struct fi_info *hints, *info;
-	int ret, result, server_ready = 0;
+	int ret = 0;
```

Comment:
Remove assignment of ret = 0.  It's set in the next call.  Use a common error path, similar to what was mentioned above.